### PR TITLE
backport: merge bitcoin#23499, #23832, #24136, #24265, #23819, #23595, #24486, #15423, #24784, #24956, #25016, #25060 (auxiliary backports: part 21)

### PR DESCRIPTION
**Pull request by kwvg**
## Breaking Changes

None expected.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by PastaPastaPasta — APPROVED** [review]
utACK b9177a374401df301fafdeaf88a8559109201ef1

File: src/validation.cpp [hunk 1]
```diff
@@ -2307,12 +2307,12 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
...
 
     TRACE6(validation, block_connected,
-        block.GetHash().data(),
+        block_hash.data(),
         pindex->nHeight,
         block.vtx.size(),
         nInputs,
         nSigOps,
-        GetTimeMicros() - nTimeStart // in microseconds (µs)
+        nTime5 - nTimeStart // in microseconds (µs)
```
Comment (UdjinM6) [inline src/validation.cpp:2315]: 23819: we have a couple more `nTime`s
```suggestion
        nTime8 - nTimeStart // in microseconds (µs)
```

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -1818,24 +1817,28 @@ RPCHelpMan getblockchaininfo()
...
+                         Consensus::DEPLOYMENT_V19,
+                         Consensus::DEPLOYMENT_V20,
+                         Consensus::DEPLOYMENT_MN_RR }) {
+        SoftForkDescPushBack(&tip, softforks, consensusParams, deploy);
+    }
+    for (auto ehf_deploy : { /* sorted by activation block */
+                             Consensus::DEPLOYMENT_WITHDRAWALS,
+                             Consensus::DEPLOYMENT_TESTDUMMY }) {
+        SoftForkDescPushBack(&tip, ehfSignals, softforks, consensusParams, ehf_deploy);
+    }
```
Comment (UdjinM6) [inline src/rpc/blockchain.cpp:1842]: 24956: nit: unrelated refactoring

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/validation.cpp (hunk 1 above)
Comment (kwvg) [reply src/validation.cpp]: Resolved in latest push!

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (kwvg) [reply src/rpc/blockchain.cpp:1842]: Split off into separate commit

**Review by UdjinM6 — APPROVED** [review]
utACK 02320032c7c779a04a8e7347b48af368a56a32cf

**Pull request merged**